src/main/java/org/quiltmc/loader/api/QuiltLoader.java

```diff
@@ -0,0 +1,191 @@
+	 * either a string, or an object. An object declaration
+	 * is of this form:<pre><blockquote>
+	 *   {
+	 *     "adapter": "&lt;a custom adatper&gt;"
+	 *     "value": "&lt;an arbitrary notation&gt;"
+	 *   }
+	 * </blockquote></pre>
+	 * A string declaration {@code <an arbitrary notation>} is equivalent to
+	 * <pre><blockquote>
+	 *   {
+	 *     "adapter": "default"
+	 *     "value": "&lt;an arbitrary notation&gt;"
+	 *   }
+	 * </blockquote></pre>
+	 * where the {@code default} adapter is the {@linkplain LanguageAdapter adapter}
+	 * offered by Fabric Loader. </p>
+	 *
+	 * @param key  the key in entrypoint declaration in {@code fabric.mod.json}
+	 * @param type the type of entrypoints
+	 * @param <T>  the type of entrypoints
+	 * @return the entrypoint containers related to this key
+	 * @throws EntrypointException if a problem arises during entrypoint creation
+	 * @see LanguageAdapter
+	 */
+	<T> List<EntrypointContainer<T>> getEntrypointContainers(String key, Class<T> type);
+
+	/**
+	 * Get the current mapping resolver.
+	 *
+	 * <p>When performing reflection, a mod should always query the mapping resolver for
+	 * the remapped names of members than relying on other heuristics.</p>
+	 *
+	 * @return the current mapping resolver instance
+	 * @since 0.4.1
+	 */
+	MappingResolver getMappingResolver();
+
+	/**
+	 * Gets the container for a given mod.
+	 *
+	 * @param id the ID of the mod
+	 * @return the mod container, if present
+	 */
+	Optional<ModContainer> getModContainer(String id);
+
+	/**
+	 * Gets all mod containers.
+	 *
+	 * @return a collection of all loaded mod containers
+	 */
+	Collection<ModContainer> getAllMods();
+
+	/**
+	 * Checks if a mod with a given ID is loaded.
+	 *
+	 * @param id the ID of the mod, as defined in {@code fabric.mod.json}
+	 * @return whether or not the mod is present in this Fabric Loader instance
+	 */
+	boolean isModLoaded(String id);
+
+	/**
+	 * Checks if Fabric Loader is currently running in a "development"
+	 * environment. Can be used for enabling debug mode or additional checks.
+	 *
+	 * <p>This should not be used to make assumptions on certain features,
+	 * such as mappings, but as a toggle for certain functionalities.</p>
+	 *
+	 * @return whether or not Loader is currently in a "development"
+	 * environment
+	 */
+	boolean isDevelopmentEnvironment();
+
+	/**
+	 * Get the current environment type.
+	 *
+	 * @return the current environment type
+	 */
+	EnvType getEnvironmentType();
+
+	/**
+	 * Get the current game instance. Can represent a game client or
+	 * server object. As such, the exact return is dependent on the
+	 * current environment type.
+	 *
+	 * <p>The game instance may not always be available depending on the game version and {@link EnvType environment}.
+	 *
+	 * @return A client or server instance object
+	 * @deprecated This method is experimental and it's use is discouraged.
+	 */
+	@Nullable
+	@Deprecated
+	Object getGameInstance();
```

Comment:
Since not every game can fulfill this and Minecraft has changed recently to launch it's server statically, there isn't truly a `game instance` any more on both sides I think we should remove this API.

Reply:
I don't think that's a good enough reason to remove this method - since it's still useful *some of the time* just not *all* of the time. Plus, this is already documented in it's javadoc, and it's marked as `Deprecated`.